src/App.jsx

```diff
@@ -14,57 +15,79 @@
+const Good = memo(
+  ({ name = '', changeGood = () => {}, isActive = false }) => {
+    const goodClass = isActive ? 'has-background-success-light' : '';
+    const buttonClass = isActive ? 'button is-info' : 'button';
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
+    return (
+      <tr data-cy="Good" className={goodClass}>
+        <td>
+          <button
+            data-cy={isActive ? 'RemoveButton' : 'AddButton'}
+            type="button"
+            className={buttonClass}
+            onClick={() => changeGood(name)}
+          >
+            {isActive ? '-' : '+'}
+          </button>
+        </td>
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+        <td data-cy="GoodTitle" className="is-vcentered">
+          {name}
+        </td>
+      </tr>
+    );
+  },
+  (prevProps, nextProps) => prevProps.isActive === nextProps.isActive,
+);
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+const GoodsBoody = ({
+  goodsList = [],
+  changeGood = () => {},
+  activeGood = '',
+}) => (
+  <tbody>
+    {goodsList.map(el => (
+      <Good
+        name={el}
+        changeGood={changeGood}
+        isActive={activeGood === el}
+        key={el}
+      />
+    ))}
+  </tbody>
+);
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+export const App = () => {
+  const [good, setGood] = useState('Jam');
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+  const selectGood = useCallback(
+    name => (good === name ? setGood(null) : setGood(name)),
+    [good],
+  );
 
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {!good ? 'No goods selected' : `${good} is selected`}
+        {good && (
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={() => setGood(null)}
+          />
+        )}
+      </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+      <table className="table">
+        <GoodsBoody
```

Comment:
Ensure that the component name `GoodsBoody` is corrected to `GoodsBody` here as well to maintain consistency and avoid potential errors.